R/runs.R

```diff
@@ -40,34 +40,86 @@ assert_new_col_length <- function(x, metrics) {
   )
 }
 
+exists_metric <- function(metric_key, run_id, client) {
+  tryCatch(
+    {
+      get_metric_history(
+        metric_key = metric_key,
+        run_id = run_id,
+        client = client
+      )
+
+      return(TRUE)
+    },
+    error = function(e) {
+      return(FALSE)
+    }
+  )
+}
+
+get_most_recent_step <- function(metric_history) {
+  max(metric_history$step, na.rm = TRUE)
+}
+
+increment_metric_step <- function(metric_key, run_id, client) {
+  get_metric_history(
+    metric_key = metric_key,
+    run_id = run_id,
+    client = client
+  ) %>%
+    get_most_recent_step() %>%
+    add(1) %>%
+    as.integer()
+}
+
 #' Log Metrics
 #'
 #' Logs a metric for a run. Metrics key-value pair that records a single float measure.
 #'   During a single execution of a run, a particular metric can be logged several times.
 #'   The MLflow Backend keeps track of historical metric values along two axes: timestamp and step.
 #'
 #' @param ... variable names from which a data.frame with `key` and `value` columns will be created.
-#' @param timestamp Timestamp at which to log the metric. Timestamp is rounded to the nearest
-#'  integer. If unspecified, the number of milliseconds since the Unix epoch is used.
-#' @param step Step at which to log the metric. Step is rounded to the nearest integer. If
-#'  unspecified, the default value of zero is used.
 #' @param client An MLFlow client. Defaults to `NULL` and will be auto-generated.
 #' @param run_id A run uuid. Automatically inferred if a run is currently active.
 #'
 #' @importFrom rlang maybe_missing
+#' @importFrom purrr imap_int
+#' @importFrom magrittr add
 #'
 #' @export
-log_metrics <- function(..., timestamp = NA, step = NA, run_id = get_active_run_id(), client = mlflow_client()) {
+log_metrics <- function(..., run_id = get_active_run_id(), client = mlflow_client()) {
 
   metrics <- get_key_value_df(...)
 
-  assert_new_col_length(timestamp, metrics)
```

Comment:
so we can get rid of `assert_new_col_length` now, right?

Reply:
it's in some tests, so if we want to keep it, maybe just define it in the test file?

Reply:
Oh yeah lemme check